aria/appilcation_tools.py

```diff
@@ -0,0 +1,257 @@
+
+    def create_blueprint_storage(self, source, main_file_name=None):
+        """
+        create blueprint model & resource
+        """
+        assert self.blueprint_path and self.blueprint_id
+        assert hasattr(self.resource_storage, 'blueprint')
+        assert hasattr(self.model_storage, 'blueprint')
+
+        self.logger.debug('creating blueprint resource storage entry')
+        self.resource_storage.blueprint.upload(
+            entry_id=self.blueprint_id,
+            source=os.path.dirname(source))
+        self.logger.debug('created blueprint resource storage entry')
+
+        self.logger.debug('creating blueprint model storage entry')
+        now = datetime.now()
+        blueprint = self.model_storage.blueprint.model_cls(
+            plan=self.blueprint_plan,
+            id=self.blueprint_id,
+            description=self.blueprint_plan.get('description'),
+            created_at=now,
+            updated_at=now,
+            main_file_name=main_file_name or 'blueprint.yaml'
+        )
+        self.model_storage.blueprint.store(blueprint)
+        self.logger.debug('created blueprint model storage entry')
+
+    def create_nodes_storage(self):
+        """
+        create nodes model
+        """
+        assert hasattr(self.model_storage, 'node')
+        assert hasattr(self.model_storage, 'relationship')
+
+        for node in copy(self.blueprint_plan['nodes']):
+            for field in ('name',
+                          'deployment_plugins_to_install',
+                          'interfaces',
+                          'instances'):
+                node.pop(field)
+            scalable = node.pop('capabilities')['scalable']['properties']
+            for index, relationship in enumerate(node['relationships']):
+                relationship = self.model_storage.relationship.model_cls(**relationship)
+                self.model_storage.relationship.store(relationship)
+                node['relationships'][index] = relationship
+
+            node = self.model_storage.node.model_cls(
+                blueprint_id=self.blueprint_id,
+                planned_number_of_instances=scalable['current_instances'],
+                deploy_number_of_instances=scalable['default_instances'],
+                min_number_of_instances=scalable['min_instances'],
+                max_number_of_instances=scalable['max_instances'],
+                number_of_instances=scalable['current_instances'],
+                **node)
+            self.model_storage.node.store(node)
+
+    def create_deployment_storage(self):
+        """
+        create deployment model & resource
+        """
+        assert self.deployment_id and self.deployment_plan
+
+        assert hasattr(self.resource_storage, 'blueprint')
+        assert hasattr(self.resource_storage, 'deployment')
+        assert hasattr(self.model_storage, 'deployment')
+
+        self.logger.debug('creating deployment resource storage entry')
+        temp_dir = tempfile.mkdtemp()
```

Comment:
please check if we can write this like this:
with tempfile.NamedTemporaryFile(mode='ab',..., delete=True) as temp_file:
    self.resource_storage.blueprint.download(entry_id=self.blueprint_id, destination=temp_file)
    self.resource_storage.deployment.upload(entry_id=self.deployment_id,  source=temp_file)
but with directory temp path...